### PR TITLE
Add vectorized password function.

### PR DESCRIPTION
**Pull request by TommyCpp**
Signed-off-by: zhongyang.wu <zhongyang.wu@outlook.com>

### What problem does this PR solve?

Add vectorized Password function

### What is changed and how it works?

What's Changed:
Add vectorized function `password`
### Check List 

Tests 

- Unit test

### Release note

### CODE REVIEW

**Comment by skyzh** [comment]
Hi @TommyCpp ! Recently I've refactored the copr framework. Now vectorized functions only accept `BytesRef`. Please refer to latest master for some examples :)

File: components/tidb_query_vec_expr/src/impl_encryption.rs [hunk 1]
```diff
@@ -59,6 +59,28 @@ pub fn sha2(
     }
 }
 
+// https://dev.mysql.com/doc/refman/5.7/en/password-hashing.html
+#[rpn_fn(capture = [ctx])]
+#[inline]
+pub fn password(ctx: &mut EvalContext, input: &Option<Bytes>) -> Result<Option<Bytes>> {
```
Comment (TennyZhuang) [inline components/tidb_query_vec_expr/src/impl_encryption.rs:65]: Please change the signature for latest coprocessor framework

**Comment by iosmanthus** [comment]
REST LGTM! @TennyZhuang PTAL

**Comment by TennyZhuang** [comment]
@breeswish 来merge

**Pull request merged**